core/server/worker/src/test/java/alluxio/worker/block/CacheRequestManagerTest.java

```diff
@@ -0,0 +1,168 @@
+    mCacheRequestManager.submitRequest(request);
+    verify(mBlockWorker).createBlock(Sessions.ASYNC_CACHE_WORKER_SESSION_ID, blockId, 0, "",
+        blockLength);
+    verify(mBlockWorker).createBlockWriter(Sessions.ASYNC_CACHE_WORKER_SESSION_ID, blockId);
+    verify(mBlockWorker).commitBlock(Sessions.ASYNC_CACHE_WORKER_SESSION_ID, blockId, false);
+  }
+
+  @Test
+  public void submitAsyncRequestCacheBlockFromUfs() throws Exception {
+    String localWorkerHostname = NetworkAddressUtils.getLocalHostName(
+        (int) ServerConfiguration.getMs(PropertyKey.NETWORK_HOST_RESOLUTION_TIMEOUT_MS));
+    int port = mRandom.nextInt(5000);
+    long blockId = mRandom.nextLong();
+    long blockLength = mRandom.nextInt(5000);
+    Protocol.OpenUfsBlockOptions openUfsBlockOptions = Protocol.OpenUfsBlockOptions.newBuilder()
+        .setMaxUfsReadConcurrency(10).setUfsPath("/a").build();
+    CacheRequest request = CacheRequest.newBuilder().setBlockId(blockId).setLength(blockLength)
+        .setOpenUfsBlockOptions(openUfsBlockOptions).setSourceHost(localWorkerHostname)
+        .setSourcePort(port).setAsync(true).build();
+    BlockReader mockBlockReader = mock(BlockReader.class);
+    when(mBlockWorker.createUfsBlockReader(Sessions.ASYNC_CACHE_UFS_SESSION_ID, blockId, 0, false,
+        openUfsBlockOptions)).thenReturn(mockBlockReader);
+    mCacheRequestManager.submitRequest(request);
+    verify(mBlockWorker, timeout(30000)).createUfsBlockReader(Sessions.ASYNC_CACHE_UFS_SESSION_ID,
+        blockId, 0, false, openUfsBlockOptions);
+  }
+
+  @Test
+  public void submitAsyncRequestCacheBlockFromRemoteWorker() throws Exception {
+    String localWorkerHostname = NetworkAddressUtils.getLocalHostName(
+        (int) ServerConfiguration.getMs(PropertyKey.NETWORK_HOST_RESOLUTION_TIMEOUT_MS));
+    String fakeRemoteWorker = localWorkerHostname + "1";
+    int port = mRandom.nextInt(5000);
+    long blockId = mRandom.nextLong();
+    long blockLength = mRandom.nextInt(5000);
+    Protocol.OpenUfsBlockOptions options = Protocol.OpenUfsBlockOptions.newBuilder()
+        .setMaxUfsReadConcurrency(10).setUfsPath("/a").build();
+    CacheRequest request = CacheRequest.newBuilder().setBlockId(blockId).setLength(blockLength)
+        .setOpenUfsBlockOptions(options).setSourceHost(fakeRemoteWorker).setSourcePort(port)
+        .setAsync(true).build();
+    setupMockReaderWriter(fakeRemoteWorker, port, blockId, blockLength, options);
+    mCacheRequestManager.submitRequest(request);
+    verify(mBlockWorker, timeout(30000)).createBlock(Sessions.ASYNC_CACHE_WORKER_SESSION_ID,
+        blockId, 0, "", blockLength);
+    verify(mBlockWorker, timeout(30000)).createBlockWriter(Sessions.ASYNC_CACHE_WORKER_SESSION_ID,
+        blockId);
+    verify(mBlockWorker, timeout(30000)).commitBlock(Sessions.ASYNC_CACHE_WORKER_SESSION_ID,
+        blockId, false);
+  }
+
+  private void setupMockReaderWriter(String source, int port, long blockId, long blockLength,
```

Comment:
yeah agree. But like I mentioned in a previous comment, it's not obvious to me how to deal with remote host case in the Integration tests and it's heavyweight to cover it in overmind. So that's why I have to put a lot of mock here to mock the RemoteBlockReader. Is there a example for testing remote host so I can try to follow that?